src/peer_digest.cc

```diff
@@ -485,13 +483,13 @@ peerDigestFetchReply(void *data, char *buf, ssize_t size)
             /* fetch->entry->mem_obj->request = nullptr; */
 
             if (!fetch->pd->cd) {
-                peerDigestFetchAbort(fetch, buf, "304 without the old in-memory digest");
+                finishAndDeleteFetch(fetch, "304 without the old in-memory digest", true);
                 return -1;
             }
 
             // stay with the old in-memory digest
-            peerDigestFetchStop(fetch, buf, "Not modified");
-            fetch->state = DIGEST_READ_DONE;
```

Comment:
We can also completely remove DIGEST_READ_DONE because it became unused.

Reply:
Nice. Please post an official PR that removes DIGEST_READ_DONE as unused and fixes this use-after-free bug. We will remove these wrappers (i.e. post this PR for official review) after that PR merges.